internal/stacks/stackruntime/apply_test.go

```diff
@@ -3110,16 +3117,14 @@ func TestApplyWithChangedInputValues(t *testing.T) {
 
 	go Apply(ctx, &applyReq, &applyResp)
 	applyChanges, applyDiags := collectApplyOutput(applyChangesCh, diagsCh)
-	if len(applyDiags) != 1 {
-		t.Fatalf("expected exactly two diagnostics, got %s", applyDiags.ErrWithWarnings())
-	}
 
 	sort.SliceStable(applyDiags, diagnosticSortFunc(applyDiags))
 	expectDiagnosticsForTest(t, applyDiags,
 		expectDiagnostic(
 			tfdiags.Error,
 			"Inconsistent value for input variable during apply",
 			"The value for non-ephemeral input variable \"input\" was set to a different value during apply than was set during plan. Only ephemeral input variables can change between the plan and apply phases."),
+		expectDiagnostic(tfdiags.Error, "Invalid inputs for component", "Invalid input variable definition object: attribute \"input\": string required."),
```

Comment:
This is the extra diagnostic discussed in the description. This is the diagnostic that now prevents the bad component from actually applying.